File: Assets/Script/Dev/Mobile/MascotModel.ts
import { MASCOT_CELL_SIZE, MASCOT_FRAME_PIVOT_X, MASCOT_PIVOT_Y } from "Dev/Mobile/MascotFrames";
export { MASCOT_CELL_SIZE, MASCOT_PIVOT_Y } from "Dev/Mobile/MascotFrames";
export const MASCOT_FRAME_COUNT = 4;

export const mascotFramePivotX = (row: number, frame: number) => MASCOT_FRAME_PIVOT_X[row * MASCOT_FRAME_COUNT + frame];

export const mascotLayout = (size: number) => {
	// Preserve the requested UI size, without pixel-art quantization.
	const scale = size / MASCOT_CELL_SIZE;
	return {
		scale,
		width: MASCOT_CELL_SIZE * scale,
		feetY: (MASCOT_CELL_SIZE / 2 - MASCOT_PIVOT_Y) * scale,
	};
};

export const mascotFrameAt = (elapsed: number, interval: number) => math.floor(elapsed / interval) % MASCOT_FRAME_COUNT;
export const mascotAnimationTime = (elapsed: number, dt: number, reducedMotion: boolean) => reducedMotion ? 0 : elapsed + dt;

File: Assets/Script/Dev/Mobile/MascotModel.lua
-- [ts]: MascotModel.ts
local ____exports = {} -- 1
local ____MascotFrames = require("Dev.Mobile.MascotFrames") -- 1
local MASCOT_CELL_SIZE = ____MascotFrames.MASCOT_CELL_SIZE -- 1
local MASCOT_FRAME_PIVOT_X = ____MascotFrames.MASCOT_FRAME_PIVOT_X -- 1
local MASCOT_PIVOT_Y = ____MascotFrames.MASCOT_PIVOT_Y -- 1
do -- 1
	local ____MascotFrames = require("Dev.Mobile.MascotFrames") -- 2
	____exports.MASCOT_CELL_SIZE = ____MascotFrames.MASCOT_CELL_SIZE -- 2
	____exports.MASCOT_PIVOT_Y = ____MascotFrames.MASCOT_PIVOT_Y -- 2
end -- 2
____exports.MASCOT_FRAME_COUNT = 4 -- 3
____exports.mascotFramePivotX = function(row, frame) return MASCOT_FRAME_PIVOT_X[row * ____exports.MASCOT_FRAME_COUNT + frame + 1] end -- 5
____exports.mascotLayout = function(size) -- 7
	local scale = size / MASCOT_CELL_SIZE -- 9
	return {scale = scale, width = MASCOT_CELL_SIZE * scale, feetY = (MASCOT_CELL_SIZE / 2 - MASCOT_PIVOT_Y) * scale} -- 10
end -- 7
____exports.mascotFrameAt = function(elapsed, interval) return math.floor(elapsed / interval) % ____exports.MASCOT_FRAME_COUNT end -- 17
____exports.mascotAnimationTime = function(elapsed, dt, reducedMotion) return reducedMotion and 0 or elapsed + dt end -- 18
return ____exports -- 18